arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -0,0 +1,1712 @@
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"name": "exportDevicesRequest",
+					"in": "body",
+					"required": true,
+					"schema": {
+						"$ref": "#/definitions/ExportDevicesRequest"
+					}
+				}, {
+					"name": "subscriptionId",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"name": "resourceGroupName",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"$ref": "#/parameters/api-version"
+				}],
+				"responses": {
+					"200": {
+						"description": "OK",
+						"schema": {
+							"$ref": "#/definitions/JobResponse"
+						}
+					},
+					"default": {
+						"description": "DefaultErrorResponse",
+						"schema": {
+							"$ref": "#/definitions/ErrorDetails"
+						}
+					}
+				},
+				"deprecated": false
+			}
+		},
+		"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/IotHubs/{resourceName}/importDevices": {
+			"post": {
+				"tags": ["POST"],
+				"summary": "Import all the Devices in an IotHub",
+				"description": "Import all the Devices in an IotHub",
+				"operationId": "IotHubResource_ImportDevices",
+				"consumes": ["application/json", "text/json"],
+				"produces": ["application/json", "text/json"],
+				"parameters": [{
+					"name": "resourceName",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"name": "importDevicesRequest",
+					"in": "body",
+					"required": true,
+					"schema": {
+						"$ref": "#/definitions/ImportDevicesRequest"
+					}
+				}, {
+					"name": "subscriptionId",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"name": "resourceGroupName",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"$ref": "#/parameters/api-version"
+				}],
+				"responses": {
+					"200": {
+						"description": "OK",
+						"schema": {
+							"$ref": "#/definitions/JobResponse"
+						}
+					},
+					"default": {
+						"description": "DefaultErrorResponse",
+						"schema": {
+							"$ref": "#/definitions/ErrorDetails"
+						}
+					}
+				},
+				"deprecated": false
+			}
+		}
+	},
+	"definitions": {
+		"Object": {
+			"type": "object",
+			"properties": {}
+		},
+		"SharedAccessSignatureAuthorizationRule": {
+			"type": "object",
+			"properties": {
+				"keyName": {
+					"type": "string"
+				},
+				"primaryKey": {
+					"type": "string"
+				},
+				"secondaryKey": {
+					"type": "string"
+				},
+				"rights": {
+					"enum": ["RegistryRead", "RegistryWrite", "ServiceConnect", "DeviceConnect"],
+					"type": "string"
+				}
+			}
+		},
+		"IotHubProperties": {
+			"type": "object",
+			"properties": {
+				"authorizationPolicies": {
+					"type": "array",
+					"items": {
+						"$ref": "#/definitions/SharedAccessSignatureAuthorizationRule"
+					}
+				},
+				"hostName": {
+					"type": "string"
+				},
+				"eventHubEndpoints": {
```

Comment:
To me it looks like they should be references. The way it is modeled currently will generate dictionaries of the referenced objects.